File: languages/div.go
package main

import (
    "errors"
    "fmt"
)

func div(x, y float64) (float64, error) {
    if y == 0 {
        return 0, errors.New("division by zero")
    }
    return x / y, nil
}

func main() {
    x, y := 1., 0.
    if z, err := div(x, y); err != nil {
        fmt.Printf("Division failed: %v\n", err)
    } else {
        fmt.Printf("The result is: %v\n", z)
    }
}

File: languages/div.js
class DivError extends Error {
    constructor(message = "division by zero") {
        super(message);
        this.name = "DivError";
    }
}

function div(x, y) {
    if (y === 0) {
        throw new DivError();
    }
    return x / y;
}

function main() {
    const x = 1.0;
    const y = 0.0;
    
    try {
        const z = div(x, y);
        console.log(`The result is: ${z}`);
    } catch (err) {
        console.error(`Division failed: ${err.message}`);
    }
}

main();